plugin/src/main/java/org/opennms/velocloud/connections/ConnectionManager.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.velocloud.connections;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import org.opennms.integration.api.v1.scv.Credentials;
+import org.opennms.integration.api.v1.scv.SecureCredentialsVault;
+import org.opennms.integration.api.v1.scv.immutables.ImmutableCredentials;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+
+public class ConnectionManager {
+
+    private static final String ATTR_ENTERPRISE_ID = "enterpriseId";
+
+    private final SecureCredentialsVault vault;
+
+    public ConnectionManager(final SecureCredentialsVault vault) {
+        this.vault = Objects.requireNonNull(vault);
+    }
+
+    /**
+     * Returns a list of all available connection aliases.
+     *
+     * @return the list of aliases
+     */
+    public Set<String> getAliases() {
+        return this.vault.getAliases();
+    }
+
+    /**
+     * Returns a connection config for the given alias.
+     *
+     * @param alias the alias of the connection config to retrieve
+     * @return The connection config or {@code Optional#empty()} of no such alias exists
+     * @throws ConnectionValidationError if the connection config is invalid
+     */
+    public Optional<Connection> getConnection(final String alias) throws ConnectionValidationError {
+        final var credentials = this.vault.getCredentials(alias);
+        if (credentials == null) {
+            return Optional.empty();
+        }
+
+        return Optional.of(new ConnectionImpl(alias, credentials));
+    }
+
+    private class ConnectionImpl implements Connection {
+        private final String alias;
+        private String apiKey;
+        private String orchestratorUrl;
+        private UUID enterpriseId;
+
+        private ConnectionImpl(final String alias, final Credentials credentials) throws ConnectionValidationError {
+            this.alias = Objects.requireNonNull(alias);
+
+            if (Strings.isNullOrEmpty(credentials.getUsername())) {
```

Comment:
Do Credentials username/password required by the SecureVault?
I wonder ( based on Jesse comment this morning), if we may have the username/password as optional in case clients don't want to use add those and just add the apikey instead.

Reply:
It looks like we're saving the orchestratorUrl as the username and the apiKey as the password. The actual username/password shouldn't get past the karaf command / rest endpoint if the plan is to just use them to generate an api key.

Reply:
> It looks like we're saving the orchestratorUrl as the username and the apiKey as the password. The actual username/password shouldn't get past the karaf command / rest endpoint if the plan is to just use them to generate an api key.

right, I would make the username:password auth only part of the `add-connection` shell command

Reply:
Main concern is, that creating an API key is expensive and these keys will never go away again